spec/session_spec.coffee

```diff
@@ -508,6 +509,32 @@ describe 'Session', ->
 
             expect(@biskoto.get(@basic_meta_cookie_name)).to.deep.equal metadata
 
+        context 'and shop_cookie_policy is specified as basic', ->
+          beforeEach ->
+            @shop_cookie_policy = 'basic'
+
+          it 'sets the session first-party basic cookie value', ->
+            metadata = { app_type: @metadata.app_type, cp: 'b', tags: @metadata.tags }
+            @stub_params({ session: @analytics_session, metadata: metadata })
+
+            @init()
+
+            expect(@biskoto.get(@basic_session_cookie_name).session).to.equal @analytics_session
+
+          it 'sets the metadata first-party basic cookie value', ->
+            metadata = { app_type: @metadata.app_type, cp: 'b', tags: @metadata.tags }
+            @stub_params({ session: @analytics_session, metadata: metadata })
+
+            @init()
+
+            expect(@biskoto.get(@basic_meta_cookie_name)).to.deep.equal metadata
+
+          it 'does not sets the session first-party full cookie value', ->
+            expect(@biskoto.get(@session_cookie_name)).to.be.null
+
+          it 'does not sets the metadata first-party full cookie value', ->
+            expect(@biskoto.get(@meta_cookie_name)).to.be.null
```

Comment:
The init method is not called and that's why the specs pass. If we call init the specs fail.

Reply:
Also if we call the init method the metadata cookie policy must be full